hack/generator/pkg/astmodel/type.go

```diff
@@ -51,3 +53,26 @@ func TypeEquals(left, right Type) bool {
 
 	return left.Equals(right)
 }
+
+// CodeGenerationPanic is a panic thrown at codegen time
+// this should only be caused by bugs
+type CodeGenerationPanic struct {
+	errors []string
+}
+
+func (p CodeGenerationPanic) ToError() error {
+	var es []error
+	for _, e := range p.errors {
+		es = append(es, errors.New(e))
```

Comment:
we should store errors in `CodeGenerationPanic` instead of strings, so the callstack is captured at point of panic creation

Reply:
or just throw raw `error`, actually!